### PR TITLE
raise errors properly on UnityRegistry

### PR DESCRIPTION
**Pull request by vincentpierre**
### Proposed change(s)

Raising errors better when something goes wrong when trying to download

### Types of change(s)

- [ ] Bug fix
- [ ] New feature
- [ ] Code refactor
- [ ] Breaking change
- [ ] Documentation update
- [ ] Other (please describe)

### Checklist
- [ ] Added tests that prove my fix is effective or that my feature works
- [ ] Updated the [changelog](https://github.com/Unity-Technologies/ml-agents/blob/master/com.unity.ml-agents/CHANGELOG.md) (if applicable)
- [ ] Updated the [documentation](https://github.com/Unity-Technologies/ml-agents/tree/master/docs) (if applicable)
- [ ] Updated the [migration guide](https://github.com/Unity-Technologies/ml-agents/blob/master/docs/Migrating.md) (if applicable)

### CODE REVIEW

File: ml-agents-envs/mlagents_envs/registry/binary_utils.py [hunk 1]
```diff
@@ -38,10 +38,13 @@ def get_local_binary_path(name: str, url: str) -> str:
             break
         try:
             download_and_extract_zip(url, name)
-        except IOError:
-            logger.debug(
-                f"Attempt {attempt + 1} / {NUMBER_ATTEMPTS} : Failed to download"
-            )
+        except Exception:  # pylint: disable=W0702
```
Comment (ervteng) [inline ml-agents-envs/mlagents_envs/registry/binary_utils.py:41]: Is there a particular exception that is given here when it fails to download?

Comment (vincentpierre) [reply ml-agents-envs/mlagents_envs/registry/binary_utils.py:41]: A lot of things could go wrong : Could have internet problems, zip extraction problems, permission issues, IOErrors...

Comment (ervteng) [reply ml-agents-envs/mlagents_envs/registry/binary_utils.py:41]: I suppose ideally we only retry certain things, specifically internet/download problems. But that seems more difficult than it's worth

Comment (ervteng) [reply ml-agents-envs/mlagents_envs/registry/binary_utils.py:41]: Maybe we can just change the message to "Failed to download and extract"

**Pull request merged**